src/rank.rs

```diff
@@ -354,4 +451,17 @@ mod test {
         }
         assert!(TaxRank::from_str("fake_data").is_err());
     }
+
+    #[test]
+    fn test_rank_to_str() {
+        for rank in RANKS.iter() {
+            let rank_str = rank.to_string();
+            assert!(rank_str.len() > 0);
```

Comment:
Clippy will probably tell you to replace it with `assert!(!rank_str.is_empty());`